.github/workflows/discover-lean-pr-testing.yml

```
@@ -179,36 +179,38 @@ jobs:
         SUCCESSFUL_MERGES="${{ steps.execute-merges.outputs.successful_merges }}"
         FAILED_MERGES="${{ steps.execute-merges.outputs.failed_merges }}"
 
-        {
-          printf $'msg<<EOF\n'
-
-          # Report successful merges
-          if [ -n "$SUCCESSFUL_MERGES" ]; then
-            printf $'### ✅ Successfully merged branches:\n\n'
-            for MERGE_INFO in $SUCCESSFUL_MERGES; do
-              IFS='|' read -r PR_NUMBER GITHUB_DIFF BRANCH <<< "$MERGE_INFO"
-              printf $'- lean#%s ([diff](%s))\n' "$PR_NUMBER" "$GITHUB_DIFF"
-            done
-            printf $'\n'
-          else
-            printf $'No branches were successfully merged.\n\n'
-          fi
+        # Start building the message
+        MESSAGE=""
+
+        # Report successful merges
+        if [ -n "$SUCCESSFUL_MERGES" ]; then
+          MESSAGE+=$'### ✅ Successfully merged branches:\n\n'
+          for MERGE_INFO in $SUCCESSFUL_MERGES; do
+            IFS='|' read -r PR_NUMBER GITHUB_DIFF BRANCH <<< "$MERGE_INFO"
+            MESSAGE+=$(printf '- lean#%s ([diff](%s))\n' "$PR_NUMBER" "$GITHUB_DIFF")
+          done
+          MESSAGE+=$'\n'
+        else
+          MESSAGE+=$'No branches were successfully merged.\n\n'
+        fi
 
-          # Report failed merges
-          if [ -n "$FAILED_MERGES" ]; then
-            printf $'### ❌ Failed merges:\n\nThe following branches need to be merged manually:\n\n```bash\n'
-            for MERGE_INFO in $FAILED_MERGES; do
-              IFS='|' read -r PR_NUMBER GITHUB_DIFF BRANCH <<< "$MERGE_INFO"
-              printf $'# Diff: %s\n' "$GITHUB_DIFF"
-              printf $'scripts/merge-lean-testing-pr.sh %s   # %s\n\n' "$PR_NUMBER" "$BRANCH"
-            done
-            printf $'```\n'
-          else
-            printf $'All branches were successfully merged! 🎉\n'
-          fi
+        # Report failed merges
+        if [ -n "$FAILED_MERGES" ]; then
+          MESSAGE+=$'### ❌ Failed merges:\n\nThe following branches need to be merged manually:\n\n```bash\n'
+          for MERGE_INFO in $FAILED_MERGES; do
+            IFS='|' read -r PR_NUMBER GITHUB_DIFF BRANCH <<< "$MERGE_INFO"
+            MESSAGE+=$(printf '# Diff: %s\n' "$GITHUB_DIFF")
+            MESSAGE+=$(printf 'scripts/merge-lean-testing-pr.sh %s   # %s\n\n' "$PR_NUMBER" "$BRANCH")
+          done
+          MESSAGE+=$'```\n'
+        else
+          MESSAGE+=$'All branches were successfully merged! 🎉\n'
+        fi
 
-          printf "EOF"
-        } >> "$GITHUB_OUTPUT"
+        # Output the message using the correct GitHub Actions syntax
+        echo "msg<<EOF" >> "$GITHUB_OUTPUT"
+        echo "$MESSAGE" >> "$GITHUB_OUTPUT"
+        echo "EOF" >> "$GITHUB_OUTPUT"
```

```suggestion
        printf 'msg<<EOF\n%s\nEOF' "$MESSAGE" >> "$GITHUB_OUTPUT"
```
I have seen this work with a single append to GITHUB_OUTPUT and no trailing line break.

I do not know how much of this is really necessary, it was mostly trial-and-error cargo-culting.